embedded/store/immustore.go

```diff
@@ -1012,7 +1012,7 @@ func (s *ImmuStore) commit(tx *Tx, offsets []int64, ts int64, blTxID uint64) err
 			if err == ErrKeyNotFound && txe.constraint == MustExist {
 				return ErrKeyNotFound
 			}
-			if err != ErrKeyNotFound {
+			if err != nil && err != ErrKeyNotFound {
```

Comment:
Minor: maybe better to move right after `s.indexer.Get` ? That way error checking will be close to the actual method and will be a bit easier to read.

Reply:
all the lines above after Get are error checking conditions as well ...